src/klein/app.py

```diff
@@ -198,16 +200,33 @@ def index(request):
 
         @returns: decorated handler function.
         """
-        segment_count = self._segments_in_url(url) + self._subroute_segments
 
-        @named("router for '" + url + "'")
+        def _urlToString(_url):
+            return _url.asText()
+
+        if isinstance(url, string_types + (binary_type, )):
+            url = URL.fromText(url)
```

Comment:
Yeah, while presence of `asIRI()/asURI()` makes it also convenient for devs who want their apps to listen on Punycode URIs, the actual server is also accepting UTF-8 bytes ^__^
just try:
```
cat "GET /pages/大阪府大東市立四条中学校/434035363318098 HTTP/1.0\nHost: facebook.com\r\n" | nc facebook.com 80
```
you can use tcpdump/wireshark to make sure that byte sequence passing to and from server contain actual Japanese UTF-8 bytes.